lab-justin/test/note-router-test.js

```diff
@@ -0,0 +1,222 @@
+'use strict';
+
+process.env.MONGO_URI = 'mongodb://localhost/test';
+
+const expect = require('chai').expect;
+const request = require('superagent-use');
+const superPromise = require('superagent-promise-plugin');
+const port = process.env.PORT || 3000;
+const baseUrl = `http://localhost:${port}`;
+const server = require('../server');
+const noteCrud = require('../lib/note-crud');
+request.use(superPromise);
+
+describe('testing module note-router', function(){
+  before((done) => {
+    if (!server.isRunning){
+      server.listen(port , () => {
+        server.isRunning = true;
+        console.log('server up ;-)', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done) => {
+    if (server.isRunning){
+      server.close(() => {
+        server.isRunning = false;
+        console.log('server down ^^;');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+//POST
+  describe('POST /api/note with valid data', function(){
+    after((done) => {
+      noteCrud
+      .removeAllNotes()
+      .then(() => done())
+      .catch(done);
+    });
+//POST 200
+    it('should return a note', (done) => {
+      request
+      .post(`${baseUrl}/api/note`)
+      .send({name: 'test note', content: 'test data'})
+      .then((res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.name).to.equal('test note');
+        done();
+      })
+      .catch(done);
+    });
+//POST 400
+    it('should return "bad request"', (done) => {
+      request
+      .post(`${baseUrl}/api/note`)
+      // .send({})
+      .then(done)
+     .catch( err => {
+       let res = err.response;
+       expect(res.status).to.eql(400);
+       expect(res.text).to.eql('bad request');
+       done();
+     });
+    });
+  });
+//GET
```

Comment:
You probably want to include these comments in the describe blocks so that you can find them both here and when they print their output in the terminal from running the tests.